compiler/rustc_middle/src/ty/impls_ty.rs

```diff
@@ -135,8 +135,8 @@ impl<'a> HashStable<StableHashingContext<'a>> for ty::RegionKind {
             ty::RePlaceholder(p) => {
                 p.hash_stable(hcx, hasher);
             }
-            ty::ReVar(..) => {
-                bug!("StableHasher: unexpected region {:?}", *self)
+            ty::ReVar(reg_vid) => {
+                reg_vid.hash_stable(hcx, hasher);
```

Comment:
Why did this change?

Reply:
Was hitting an ICE because `ConstraintCategory` is `HashStable` and we encounter `ReVar` in the `substs` that we added.

Reply:
Wouldn't we still ICE if we added that here (due to delay_span_bug being called, but no error having been found)?

Reply:
Yes, but only if no other errors are emitted, which would only happen if a diagnostic isn't built. I'm assuming that `ReVar` is only being encountered due to diagnostic changes and not changes that would affect the final binary, right?